src/hooks/questionary/usePersistQuestionaryEditorModel.ts

```diff
@@ -224,31 +203,76 @@ export function usePersistQuestionaryEditorModel() {
             step => step.topic.id === extendedTopicId
           );
 
-          executeAndMonitorCall(() =>
-            assignQuestionsToTopic(
-              state.templateId,
-              reducedTopic!.topic.id,
-              reducedTopic!.fields.map(
-                field => field.question.proposalQuestionId
-              )
-            )
-          );
+          let destinationTopic = reducedTopic;
+
           if (reducedTopicId !== extendedTopicId) {
-            executeAndMonitorCall(() =>
-              assignQuestionsToTopic(
-                state.templateId,
-                extendedTopic!.topic.id,
-                extendedTopic!.fields.map(
-                  field => field.question.proposalQuestionId
-                )
-              )
-            );
+            destinationTopic = extendedTopic;
           }
+
+          const questionRelToChange =
+            destinationTopic?.fields[action.payload.destination.index];
+          const prevQuestion =
+            destinationTopic?.fields[action.payload.destination.index - 1];
+          const nextQuestion =
+            destinationTopic?.fields[action.payload.destination.index + 1];
```

Comment:
Yes this is handled with default values in the sort order calculations. For example if one of the values or both are `undefined` it will result with number between some defaults.